scripts/pyi_generator.py

```diff
@@ -0,0 +1,144 @@
+"""The pyi generator module."""
+
+import importlib
+import inspect
+import os
+from inspect import getfullargspec
+from pathlib import Path
+from typing import Any
+
+import black
+
+from reflex.components.component import Component
+
+
+def _get_type_hint(value):
+    res = ""
+    if hasattr(value, "_name") and value._name == "Optional":
+        res = value.__args__[0].__name__
+    elif isinstance(value, str):
+        res = value
+    else:
+        res = value.__name__
+    return res
```

Comment:
```suggestion
def _get_type_hint(value):
    res = ""
    args = get_args(value)
    if args:
        res = f"{value.__name__}[{', '.join([_get_type_hint(arg) for arg in args if arg is not type(None)])}]"
        if value.__name__ == "Var":
            res = " | ".join([res] + [_get_type_hint(arg) for arg in args if arg is not type(None)])
    elif isinstance(value, str):
        res = value
    else:
        res = value.__name__
    return res
```

If we call this recursively on the args of the type annotation, then we can keep the `Var[bool]`, etc and also get the union functionality of acception `Var[bool] | bool`.

Looking through these generated pyi files though, i'm realizing we have some discrepancies, some of which need to be fixed in the main package, but at least for the hints, i think we do need default values of `None` everywhere, which means these generated annotations basically all need to be `Optional`.

Reply:
I don't think we can use your suggestion as-is because the | is not available in python 3.7.

It's a good idea however, I'll write something along those line to make it work and to handle the defaults to `None`